services/geocoding-v6.js

```diff
@@ -0,0 +1,231 @@
+ * @param {String} [config.street] - The name of the street in the address (Stuctured Input specific field)
+ * @param {String} [config.place] - Typically these are cities, villages, municipalities, etc. (Stuctured Input specific field)
+ * @param {String} [config.region] - Top-level sub-national administrative features, such as states in the United States or provinces in Canada or China. (Stuctured Input specific field)
+ * @param {String} [config.country] - The number associated with the house (Stuctured Input specific field)
+ * @param {String} [config.postcode] - Postal codes used in country-specific national addressing systems. (Stuctured Input specific field)
+ * @param {String} [config.locality] - Official sub-city features (Stuctured Input specific field)
+ * @param {boolean} [config.autocomplete=true] - Return autocomplete results or not.
+ * @param {String} [config.worldview="us"] - Filter results to geographic features whose characteristics are defined differently by audiences belonging to various regional, cultural, or political groups.
+ * @return {MapiRequest}
+ *
+ * @example
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   limit: 2
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ * 
+ * @example
+ * // geocoding in structured input mode
+ * geocodingClient.forwardGeocode({
+ *   mode: 'structured',
+ *   address_number: '12',
+ *   street: 'Main str.'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * @example
+ * // geocoding with proximity
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   proximity: [-95.4431142, 33.6875431]
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * // geocoding with countries
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   countries: ['fr']
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * // geocoding with bounding box
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   bbox: [2.14, 48.72, 2.55, 48.96]
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ */
+GeocodingV6.forwardGeocode = function(config) {
+  v.assertShape({
+    query: v.required(v.string),
+    mode: v.oneOf('normal', 'structured'),
+    countries: v.arrayOf(v.string),
+    proximity: v.oneOf(v.coordinates, 'ip'),
+    types: v.arrayOf(v.oneOf(featureTypes)),
+    bbox: v.arrayOf(v.number),
+    language: v.string,
+    limit: v.number,
+    worldview: v.string,
+    autocomplete: v.boolean,
+
+    // structured input fields
+    address_number: v.string,
+    street: v.string,
+    place: v.string,
+    region: v.string,
+    country: v.string,
+    postcode: v.string,
+    locality: v.string
+  })(config);
+
+  config.mode = config.mode || 'normal';
```

Comment:
I'd propose using a different word for 'normal'. Some ideas are `'simple'`, '`standard'`